Solutions/Marain.ContentManagement.Abstractions/Marain/Cms/CompoundContent.cs

```diff
@@ -0,0 +1,51 @@
+﻿// <copyright file="CompoundContent.cs" company="Endjin Limited">
+// Copyright (c) Endjin Limited. All rights reserved.
+// </copyright>
+
+namespace Marain.Cms
+{
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// A <see cref="IContentPayload"/> representing content composed from a set of other content, without any particular layout of its own.
+    /// </summary>
+    public class CompoundContent : IContentPayload
+    {
+        /// <summary>
+        /// The registered content type for the content payload.
+        /// </summary>
+        public const string RegisteredContentType = "application/vnd.marain.cms.contentpayload.compoundcontent";
+        private List<ContentSource> children;
+
+        /// <inheritdoc/>
+        public string ContentType
+        {
+            get
+            {
+                return RegisteredContentType;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the map of ABTest group ID to <see cref="Content.Id"/>.
+        /// </summary>
+        public List<ContentSource> Children
+        {
+            get { return this.children ?? (this.children = new List<ContentSource>()); }
+            set { this.children = value; }
+        }
+
+        /// <inheritdoc/>
+        public IContentPayload Copy(bool replaceId)
+        {
+            return new CompoundContent() { Children = new List<ContentSource>(this.Children) };
```

Comment:
Yes. I have been going round on this. I need to write up the considerations and trade offs here.

I think the answer is that it *has* to be a shallow copy, because these are references to other content, and there is no meaningful sense in which they can be "copied".